### PR TITLE
Add support for adding basic auth tokens to the context

### PR DESCRIPTION
**Pull request by AntonNguyen**
I noticed that jwt transport was able to add bearer tokens to the context (https://github.com/go-kit/kit/blob/master/auth/jwt/transport.go#L47), but with basic auth, it wasn't available.

### CODE REVIEW

File: auth/basic/transport.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+}
+
+// ContextToGRPC moves a basic token from context to grpc metadata. Particularly
+// useful for clients.
+func ContextToGRPC() grpc.ClientRequestFunc {
+	return func(ctx context.Context, md *metadata.MD) context.Context {
+		token, ok := ctx.Value(BasicTokenContextKey).(string)
+		if ok {
+			// capital "Key" is illegal in HTTP/2.
+			(*md)["authorization"] = []string{generateAuthHeaderFromToken(token)}
```
Comment (peterbourgon) [inline auth/basic/transport.go:79]: This is enforced by a different layer, you don't need to assert it here.

File: auth/basic/transport.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	authHeaderParts := strings.Split(val, " ")
+	if len(authHeaderParts) != 2 || !strings.EqualFold(authHeaderParts[0], basic) {
+		return "", false
+	}
+
+	return authHeaderParts[1], true
+}
+
+func generateAuthHeaderFromToken(token string) string {
+	return fmt.Sprintf(basicFormat, token)
```
Comment (peterbourgon) [inline auth/basic/transport.go:96]: There's no reason to define this as a constant, just use a string literal here.

File: auth/basic/transport.go [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+			(*md)["authorization"] = []string{generateAuthHeaderFromToken(token)}
+		}
+
+		return ctx
+	}
+}
+
+func extractTokenFromAuthHeader(val string) (token string, ok bool) {
+	authHeaderParts := strings.Split(val, " ")
+	if len(authHeaderParts) != 2 || !strings.EqualFold(authHeaderParts[0], basic) {
```
Comment (peterbourgon) [inline auth/basic/transport.go:88]: There's no reason to define `basic` as a constant, just use a string literal.

**Review by peterbourgon — COMMENTED** [review]
Cool, can you add a `doc.go`?

**Comment by peterbourgon** [comment]
Ping :)

**Comment by AntonNguyen** [comment]
@peterbourgon Sorry for the delays - updated the PR! 😄

**Comment by AntonNguyen** [comment]
@peterbourgon Ping 👋

**Comment by AntonNguyen** [comment]
No approvals, so I'm just gonna close this.

**Pull request closed without merging**